### DISCUSSION TITLE
Drawer gets trapped in Slider overflow or misbehaves with Portal

### DISCUSSION THREAD

**Discussion by junkieshuffle** [chakra-ui/chakra-ui#9208] [Q&A] [answered]
### Description

I am encountering issues when using the Drawer component inside a slider in Chakra UI v3. Here’s the scenario:
	• I have a slider with multiple items, each containing a button that opens a Drawer.
	• Without using a portal, the Drawer gets trapped inside the slider’s overflow area, making it unusable.
	• When I enable a portal to resolve this, the “Enroll to the lottery” button unexpectedly renders outside of the intended Card.Footer, breaking the UI structure.

if we could have a property over DrawerRoot to define where to append the drawer (like body or wherever), that would have been perfect.

`

			<Card.Footer className='flex flex-col items-stretch'>
				{status === 'onprogress' && (
					<>
						<ProgressRoot mt={4} colorPalette={colorPalette} striped animated>
							<HStack gap='2'>
								{/* <ProgressLabel>10</ProgressLabel> */}
								<ProgressValueText>%10</ProgressValueText>
								<ProgressBar flex='1' />
								<ProgressValueText>250/{formatNumber(poolSize)}</ProgressValueText>
							</HStack>
						</ProgressRoot>

						<LotteryDrawerItem data={data}>
							<Button colorPalette={colorPalette}>Enroll {label} Lottery</Button>
						</LotteryDrawerItem>
					</>
				)}
				{status === 'finished' && (
					<>
						{endTime && (
							<Text textAlign={'center'} fontWeight={'bold'} fontSize={'small'}>
								Finished on {formatDateTime(endTime)}
							</Text>
						)}

						<Button variant={'surface'} colorPalette={colorPalette}>
							View Rewards
						</Button>
					</>
				)}
			</Card.Footer>
`

`

export const LotteryDrawerItem = ({
	data,
	children,
}: {
	data: LotteryItem
	children?: ReactNode
}) => {
	

	return (
		<>
			<DrawerRoot
				// open={open}
				// initialFocusEl={() => ref.current}
				// trapFocus={true}
				placement={'bottom'}
				preventScroll={false}
				closeOnInteractOutside={true}

				// onPointerDownOutside={onClose}
			>
				<DrawerBackdrop />

				<DrawerTrigger asChild>{children}</DrawerTrigger>

				<DrawerContent roundedTop={'2xl'} maxW={'25rem'} mx={'3'}>
					<DrawerCloseTrigger />
					<DrawerHeader>
						<DrawerTitle>Lottery info</DrawerTitle>
					</DrawerHeader>
					<DrawerBody></DrawerBody>
					<DrawerFooter></DrawerFooter>
				</DrawerContent>
			</DrawerRoot>
		</>
	)
}

`

Expected Behavior
	• The Drawer should render correctly and be usable, even when inside a slider with an overflow property.
	• When using a portal, the Drawer content should not break or disrupt the placement of the trigger button or other surrounding elements.

Current Behavior
	• The Drawer is confined by the slider’s overflow when not using a portal.
	• Using a portal causes UI elements (e.g., the “Enroll to the lottery” button) to render outside their intended container.

<img width="395" alt="image" src="https://github.com/user-attachments/assets/5130aa05-b16b-4c84-99f7-7c030039fa79">
<img width="387" alt="image" src="https://github.com/user-attachments/assets/166b1e8a-66fd-421b-9ece-5bc43deef146">

### Link to Reproduction

https://tonflash.vercel.app

### Steps to reproduce

1. Create a slider containing multiple items.
2. Add a button inside each item that opens a Drawer.
3. Observe the behavior with and without a portal.

### Chakra UI Version

^3.1.2

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by segunadebayo** [comment]
The fix for the code you shared above is to wrap the DrawerContent within DrawerPositioner (import it from chakra)

**Reply by junkieshuffle** [reply]
thank you @segunadebayo , However The DrawerPositioner did not fix the issue.
Also, I couldn't find any prop for the Positioner in the docs or interfaces

**Reply by junkieshuffle** [reply]
any updates regarding the issue?

**Comment by segunadebayo** [comment, marked as answer]
Wrap `Drawer.Content` in `Drawer.Positioner` and portal it. That keeps the drawer out of the slider overflow.
